pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/Utils.java

```diff
@@ -0,0 +1,103 @@
+    private static final String BATCH_SIZE_FIELD = "batchSize";
+    private static final String ENCRYPTION_CTX_FIELD = "encryptionCtx";
+
+    /**
+     * Serializes sink-record into json format. It encodes encryption-keys, encryption-param and payload in base64
+     * format so, it can be sent in json.
+     * 
+     * @param inputRecordContext
+     * @param data
+     * @return
+     */
+    public static String serializeRecordToJson(RecordContext inputRecordContext, byte[] data) {
+        if (inputRecordContext == null) {
+            return null;
+        }
+        JsonObject result = new JsonObject();
+        result.addProperty(PAYLOAD_FIELD, getEncoder().encodeToString(data));
+        if (inputRecordContext.getProperties() != null) {
+            JsonObject properties = new JsonObject();
+            inputRecordContext.getProperties().entrySet()
+                    .forEach(e -> properties.addProperty(e.getKey(), e.getValue()));
+            result.add(PROPERTIES_FIELD, properties);
+        }
+        if (inputRecordContext.getEncryptionCtx().isPresent()) {
+            EncryptionContext encryptionCtx = inputRecordContext.getEncryptionCtx().get();
+            JsonObject encryptionCtxJson = new JsonObject();
+            JsonObject keyBase64Map = new JsonObject();
+            JsonObject keyMetadataMap = new JsonObject();
+            encryptionCtx.getKeys().entrySet().forEach(entry -> {
+                keyBase64Map.addProperty(entry.getKey(), getEncoder().encodeToString(entry.getValue().getKeyValue()));
+                Map<String, String> keyMetadata = entry.getValue().getMetadata();
+                if (keyMetadata != null && !keyMetadata.isEmpty()) {
+                    JsonObject metadata = new JsonObject();
+                    entry.getValue().getMetadata().entrySet()
+                            .forEach(m -> metadata.addProperty(m.getKey(), m.getValue()));
+                    keyMetadataMap.add(entry.getKey(), metadata);
+                }
+            });
+            encryptionCtxJson.add(KEY_MAP_FIELD, keyBase64Map);
+            encryptionCtxJson.add(KEY_METADATA_MAP_FIELD, keyMetadataMap);
+            Map<String, String> metadataMap = encryptionCtx.getMetadata();
+            if (metadataMap != null && !metadataMap.isEmpty()) {
+                JsonObject metadata = new JsonObject();
+                encryptionCtx.getMetadata().entrySet().forEach(m -> metadata.addProperty(m.getKey(), m.getValue()));
+                encryptionCtxJson.add(METADATA_FIELD, metadata);
+            }
+            encryptionCtxJson.addProperty(ENCRYPTION_PARAM_FIELD,
+                    getEncoder().encodeToString(encryptionCtx.getParam()));
+            encryptionCtxJson.addProperty(ALGO_FIELD, encryptionCtx.getAlgorithm());
+            if (encryptionCtx.getCompressionType() != null) {
+                encryptionCtxJson.addProperty(COMPRESSION_TYPE_FIELD, encryptionCtx.getCompressionType().name());
```

Comment:
Do we have the compression type within encryption context?

Reply:
yes, compression-metadata is part of [EncryptionContext](https://github.com/apache/incubator-pulsar/blob/master/pulsar-common/src/main/java/org/apache/pulsar/common/api/EncryptionContext.java#L39) as decompression can happen after we decrypt the message. so, it's part of the ecnryption-ctx.